server/services/store/sqlstore/board_insights.go

```diff
@@ -0,0 +1,129 @@
+package sqlstore
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/mattermost/focalboard/server/model"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+func (s *SQLStore) getTeamBoardsInsights(db sq.BaseRunner, teamID string, duration string) ([]*model.BoardInsight, error) {
+	/**
+	Some squirrel issues to note here are
+	1. https://github.com/Masterminds/squirrel/issues/285 - since we're using 1+ sub queries. When placeholders are counted for second query, the placeholder names are repeated.
+		This is the reason to not use conditional operators in Where clauses which would eventually parametrize the variables.
+	*/
+
+	boardsHistoryQuery := s.getQueryBuilder(db).Select("boards.id", "boards.title", "count(boards_history.id) as count", "boards_history.modified_by", "boards.created_by").
+		From(s.tablePrefix + "boards_history as boards_history").
+		Join(s.tablePrefix + "boards as boards on boards_history.id = boards.id").
+		Where(fmt.Sprintf("boards_history.insert_at > %s and boards.team_id = '%s' and boards_history.modified_by != 'system' and boards.delete_at = 0",
```

Comment:
Hi @plant99 
There are no SQL sanitizations in place while using `fmt.Sprintf` statements. As a result these are vulnerable to SQL Injections. Instead please use the Query Generator offered by Squirrel client. Please check other code in sqlstore for examples. 
Thanks.

Reply:
@srkgupta Thanks for the feedback! Due to [this](https://github.com/mattermost/focalboard/pull/3005/commits/7c2d3134c734f04d6421126a20ceb74d24c5e64e#diff-456b30574cd5964ecde484e5d156d4c8878aa6274bb257588fabc073a9f72538R17) we can't use parameters in this query. So `where` can't be constructed with `sq.Conditional`.

Also I can't find any similar use-cases in focalboard repository. 

1. I'll check mattermost-server or [this](https://github.com/mattermost/mattermost-server/pull/19953/files) PR for how to sanitise fields in a manually constructed query.
2. While I find a workaround, please suggest me fixes